### ISSUE TITLE
Logged-out users are redirected to the login page on the issue page

### ISSUE THREAD

**Issue by rokroskar** [SwissDataScienceCenter/renku-ui#905]
Trying to see issue details in a public project as a logged-out user is not possible because the user is redirected to the login page. Issues should be readable by anyone in public projects.

**Comment by lorenzo-cavazzi** [comment]
The [documentation about comments](https://docs.gitlab.com/ce/api/issues.html#comments-on-issues) suggests using the [notes API](https://docs.gitlab.com/ce/api/notes.html#list-project-issue-notes) to interact with issue comments.
This is what we are doing, but anonymous users still get a `401 Unauthorized` response even for discussions that are meant to be public. I couldn't find a way to fix this problem. I opened a [ticket on GitLab](https://gitlab.com/gitlab-org/gitlab/-/issues/217369), let's see if we get any insight.

**Comment by lorenzo-cavazzi** [comment]
As a follow-up, I fear [the answer we got](https://gitlab.com/gitlab-org/gitlab/-/issues/217369#note_343217825) from one of the GitLab devs means there is no way to show the issue discussions to anonymous users.

**Comment by lorenzo-cavazzi** [comment]
We may decide to:
* Do not change this (the user simply gets redirected to the login page)
* Add a warning message, something like `you need to log in to see the discussion`. We may even suggest going to GitLab (providing the link/button) to visualize the issue. Not sure I would go that far since we would also miss a few features in GitLab that we implemented in the UI (i.e. preview notebooks)

**Comment by rokroskar** [comment]
That doesn't make any sense. I guess GitLab is not using its own API?

**Comment by lorenzo-cavazzi** [comment]
Apparently... In the GitLab UI I see a query to https://dev.renku.ch/gitlab/rokroskar/scratch-project/-/issues/1/discussions.json to get the comments

**Comment by rokroskar** [comment]
Ha! that's pretty funny actually. I guess we could do the same.

**Comment by lorenzo-cavazzi** [comment]
Yeah, we can do that, it actually works (if the project id is in the `<namespace>/<project>` format). My only fear is that this can change at any release since it's an undocumented way to get the content but... we can take the risk :D

**Comment by rokroskar** [comment]
Maybe follow up with this in the issue you opened up over there...? 🤔

**Comment by lorenzo-cavazzi** [comment]
Since we want to support anonymous navigation but we also want to limit the risk of breaking this feature, the safest way is using the undocumented API only for anonymous users
